src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final MovieSessionService movieSessionService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
+            shoppingCartDtoResponseMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService, UserService userService,
+                                  MovieSessionService movieSessionService,
+                                  DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          shoppingCartDtoResponseMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartDtoResponseMapper = shoppingCartDtoResponseMapper;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartDtoResponseMapper.toDto(shoppingCartService
+                .getByUser(userService.get(userId)));
+    }
+
+    @PutMapping("/movie-sessions")
+    public void addSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {

```

Comment:
add response

Reply:
I think that this method can be void because I use addSession() method from ShoppingCartService here and it is also void

Reply:
It can be void and can return the response body it depends on business logic.  now we don't have clear requirements, so it can be void